django/core/management/commands/migrate.py

```diff
@@ -272,6 +277,22 @@ def handle(self, *args, **options):
             ModelState.from_model(apps.get_model(*model)) for model in model_keys
         ])
 
+        if options['prune']:
+            if not options['app_label']:
+                raise CommandError('Migrations can be pruned only when an app is specified.')
+            to_prune = set(executor.loader.applied_migrations) - set(executor.loader.disk_migrations)
+            to_prune = sorted(migration for migration in to_prune if migration[0] == app_label)
+            if self.verbosity > 0:
+                self.stdout.write('Pruning migrations:', self.style.MIGRATE_HEADING)
+            if to_prune:
+                for migration in to_prune:
+                    app, name = migration
+                    if self.verbosity > 0:
+                        self.stdout.write(f'  Pruning {app}.{name}')
+                    executor.recorder.delete(app, name)
+            elif self.verbosity > 0:
+                self.stdout.write('  No migrations to prune.')
+
```

Comment:
`--prune` is ignored when `--plan` is used. Maybe we should raise an error that they're mutually exclusive.